.github/workflows/npm-publish.yml

```diff
@@ -14,6 +14,7 @@ jobs:
           node-version-file: '.nvmrc'
           cache: 'npm'
       - run: npm ci
+      - run: npm run build
```

Comment:
Do you need to do this for the tests, or do you do this as a test? 

You should also do this as a publish step, or implement it as a prepublish step.

Reply:
This is done as a test to be sure the compilation works

Reply:
Changed this, now doing type check in lint script and building js and d.ts in republish step